conda/mpi/base_build.sh

```diff
@@ -10,28 +10,38 @@ TEMP_CXXFLAGS=${CXXFLAGS//-std=c++[0-9][0-9]}
 # to the conda build process. Finally, `-std=c++17` can be appended to the end.
 ACTIVATION_CXXFLAGS=${TEMP_CXXFLAGS%%-fdebug-prefix-map*}-std=c++17
 
+# Declare our activation cppflags in preperation to have offending flags removed.
+ACTIVATION_CPPFLAGS=${CPPFLAGS}
+
+# The following flags should also be removed as per issue: https://github.com/idaholab/moose/issues/28283
+o_flags=(-D_FORTIFY_SOURCE=2)
+for strip_flag in "${o_flags[@]}"; do
+    ACTIVATION_CPPFLAGS=${ACTIVATION_CPPFLAGS//"${strip_flag}"}
+    ACTIVATION_CXXFLAGS=${ACTIVATION_CXXFLAGS//"${strip_flag}"}
+done
```

Comment:
Do these flags end up being present when conda building "downstream" packages like PETSc/libMesh? I was wondering, per our slack discussion this morning, whether we could try removing *all* the flags from `CPPFLAGS,CXXFLAGS,CFLAGS,LDFLAGS,FCFLAGS,FFLAGS`, etc. that aren't related to includes or linking libraries?

Reply:
We can, but I need to know the flags you want removed.

It's a little hacky, but as you see below in the changes, I am renaming the activation/deactivation scripts in an attempt to fire off last (and therefore take precedence).

Reply:
Hmm, I don't have a Linux conda environment to look at, but in my Mac ARM conda env, I would axe (if we haven't already)

- From `CFLAGS`: `-ftree-vectorize -O2`
- `CXXFLAGS`: `-ftree-vectorize -O2`
- `LDFLAGS`: `-Wl,-dead_strip_dylibs` (I believe I got this one already)
- `FFLAGS`: `-march=armv8.3-a -ftree-vectorize -fno-stack-protector -O2`

Reply:
Ok I'm looking at Linux now. I would axe the following:

- `CFLAGS`: `-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fno-plt -O2`
- `CXXFLAGS`: same as for `CFLAGS`
- `LDFLAGS`: `-Wl,-O2 -Wl,--allow-shlib-undefined`
- `FFLAGS`: same as for `CFLAGS`

Pinging @roystgnr to see if he agrees with me. PETSc will handle its own optimization flags if we do not force it to take some. And for libMesh optimization decisions should be made on a per-method basis. Moreover, I am skeptical of any `-m` options passed by a general distribution manager since they have no way of knowing what architecture the client is actually running on. Again, I think this decision should be handled by the client

Reply:
This will be a bit tricky, or more like hacky. We need to keep `-march -mtune` where they are at _during `conda build`_. But remove them later during end-user usage.

The reason for this is that `conda build` is building packages made for CPU arch's at or later than what they are set to. It's what allows these packages to be useful on such a broad range of hardware.

Reply:
Getting rid of `-march=nocona` would be desirable, since that way there's no way of it accidentally preempting a more up to date `-march` setting from another source (and thereby forcing the compiler to unnecessarily create "fallback" code paths for nocona compatibility).  But it isn't necessary, since `nocona` is so ancient there's no way that's going to break anything.

Getting rid of `-mtune=haswell` is similar; that will never break anything, but if someone is trying e.g. `-mtune=native` then we want them to get faster code that way, not to get faster code if their flag gets appended but not if it gets prepended.

`-ftree-vectorize -O2` could screw with debuggers.  Definitely want those out of there.

`-fPIC` is something that libtool will probably add back for us regardless, since Position-Independent Code is what can be linked into .so libraries; no harm in leaving it.

I don't actually understand `-fno-plt`.  The man page makes it seem like it might need to be set or unset for all code in the same executable, though, which worries me a lot.